kv/replica_slice_test.go

```diff
@@ -110,3 +111,22 @@ func TestReplicaSetMoveToFront(t *testing.T) {
 		t.Errorf("expected order %s, got %s", exp, stores)
 	}
 }
+
+func TestReplicaSetRandomOrder(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	rs := replicaSlice(nil)
+	for i := 0; i < 5; i++ {
+		rs = append(rs, replicaInfo{ReplicaDescriptor: roachpb.ReplicaDescriptor{StoreID: roachpb.StoreID(i + 1)}})
+	}
+
+	rs.SortRandomOrder(2)
+	expNot := []roachpb.StoreID{1, 2, 4, 5, 3}
```

Comment:
Why `1, 2, 4, 5, 3`? 

It's important to test that `SortRandomOrder` does *not* modify the first N elements of the array, not just that it does not equal some expected value. 

Shuffling the last three elements of a five-element list only has `3! = 6` possible outcomes, so this test has a 1/6 chance of failing even when the code is correct. It needs to either use a much larger list or run in a loop so that it doesn't fail on one unlucky roll.